test/test_common/environment.cc

```diff
@@ -206,4 +211,18 @@ void TestEnvironment::exec(const std::vector<std::string>& args) {
     RELEASE_ASSERT(false);
   }
 }
+
+std::string TestEnvironment::writeStringToFileForTest(const std::string& filename,
+                                                      const std::string& contents) {
+  // We rather unfortunately have to use runfilesPath becauase many consumers of
+  // these files use temporaryFileSubstitute, which assumes runfilesPath.
+  const std::string out_path = TestEnvironment::runfilesPath(filename);
```

Comment:
This seems dangerous, in general runfiles may be read-only. Is there no way to avoid this?